factory-reset/eos-factory-reset-users

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash
+
+if [ -f /etc/adduser.conf ]; then
+    . /etc/adduser.conf
+fi
+
+first_uid=${FIRST_UID:-1000}
+last_uid=${LAST_UID:-59999}
+
+IFS=':'
+while read -r user _ uid _ _ _ _; do
+    if [[ $uid -ge $first_uid && $uid -le $last_uid ]]; then
+        echo "Deleting user $user ($uid)"
+        userdel -r "$user"
+    fi
+done </etc/passwd
```

Comment:
The `/etc/passwd` that comes from the ostree only has root in it. Is there any reason why you're trying to limit the IDs?

Reply:
Alternatively, the actual "factory" ID files are in `/usr/etc/passwd`. You could compare them, remove any entries not in the factory and then just copy them over.

Furthermore, this doesn't handle `/etc/group`, `/etc/shadow` or `/etc/gshadow`. Out of the box root has no password, but someone could have set one. I think what I'd actually do:
```
# Remove non-root users and their files
while IFS=: read -r -a entry; do
    user=${entry[0]}
    if [ -z "$user" ] || [ "$user" = root ]; then
        continue
    fi
    # --remove-all-files might be too slow since it walks the entire system.
    # --remove-home would be faster
    deluser --remove-all-files "$user"
done

# Restore factory ID and password files
cp -a /usr/etc/{passwd,group,shadow,gshadow} /etc

# Wipe any backup ID and password files
rm -f /etc/{passwd,group,shadow,gshadow}-
```
I think if you want to read in both `/etc/passwd` and `/usr/etc/passwd` and compare them to make sure you're not deleting any factory users, then I'd probably get out of shell.

Reply:
> The `/etc/passwd` that comes from the ostree only has root in it. Is there any reason why you're trying to limit the IDs?
>
> Alternatively, the actual "factory" ID files are in `/usr/etc/passwd`. You could compare them, remove any entries not in the factory and then just copy them over.
>
> Furthermore, this doesn't handle `/etc/group`, `/etc/shadow` or `/etc/gshadow`. Out of the box root has no password, but someone could have set one. I think what I'd actually do:
> 
> ```
> # Remove non-root users and their files
> while IFS=: read -r -a entry; do
>     user=${entry[0]}
>     if [ -z "$user" ] || [ "$user" = root ]; then
>         continue
>     fi
>     # --remove-all-files might be too slow since it walks the entire system.
>     # --remove-home would be faster
>     deluser --remove-all-files "$user"
> done
> 
> # Restore factory ID and password files
> cp -a /usr/etc/{passwd,group,shadow,gshadow} /etc
> 
> # Wipe any backup ID and password files
> rm -f /etc/{passwd,group,shadow,gshadow}-
> ```
> I think if you want to read in both `/etc/passwd` and `/usr/etc/passwd` and compare them to make sure you're not deleting any factory users, then I'd probably get out of shell.

This is an initial version of the tool, which only addresses the requirements of T24613, which are to remove the user-created accounts so on the next boot the user is taken back to the FBE. This limited scope has been acknowledged by @cosimoc on https://phabricator.endlessm.com/T24613#668534. Its expansion to remove system users and reset other parts of the system like network config, locale, machine-id etc, is covered by T23396 and T10627. But I'll take these points for the more fully-fledged factory reset tool.

`userdel -r` and `deluser --remove-home` both take care of removing the user and groups owned by the user from all of passwd, group, shadow, gshadow.